parlai/agents/fid/fid.py

```diff
@@ -318,6 +320,56 @@ def __init__(self, opt: Opt, shared: TShared = None):
         super().__init__(opt, shared=shared)
 
 
+class GoldDocRetrieverFiDAgent(SearchQueryFiDAgent):
+    """
+    Uses the gold retrived docs (documents shown to crowdsourcing agents).
+
+    This FiD agents has a mock retriever that picks the retrieved docs from the observed
+    example.
+    """
+
+    def __init__(self, opt: Opt, shared: TShared = None):
+        opt = deepcopy(opt)
+        opt['rag_retriever_type'] = RetrieverType.OBSERVATION_ECHO_RETRIEVER.value
+        super().__init__(opt, shared=shared)
+
+    def observe(self, observation: Union[Dict, Message]) -> Message:
+        if observation.is_padding():
+            return observation
+
+        if not observation.get(consts.RETRIEVED_DOCS):
+            self.model.retriever.set_retrieve_doc(
+                retrieved_docs=None, selected_docs=None, selected_sentences=None
+            )
+        else:
+            rertrieved_docs = []
+            for doc_id, doc_title, doc_txt in zip(
+                observation[consts.RETRIEVED_DOCS_URLS],
+                observation[consts.RETRIEVED_DOCS_TITLES],
+                observation[consts.RETRIEVED_DOCS],
+            ):
+                rertrieved_docs.append(
+                    Document(docid=doc_id, title=doc_title, text=doc_txt)
+                )
+
+            selected_docs = []
+            for doc_id, doc_title, doc_txt in zip(
+                observation[consts.SELECTED_DOCS_URLS],
+                observation[consts.SELECTED_DOCS_TITLES],
+                observation[consts.SELECTED_DOCS],
+            ):
+                selected_docs.append(
+                    Document(docid=doc_id, title=doc_title, text=doc_txt)
+                )
```

Comment:
yeah, although as we've discussed this may be better to handle on the teacher side --> i.e., this retriever can be extremely simple and ignorant, such that it just uses the documents provided by the teacher as is